CONTRIBUTING.md

```diff
@@ -82,6 +82,27 @@ some bugs or smaller features.  We have a [feature development
 process](https://github.com/kubernetes/features/blob/master/README.md), but
 navigating the Kubernetes system as a newcomer can be very challenging.
 
+### Building the project
+
+There are a few things you need to build and test Kubernetes:
+1. `make` - the human interface to the Kubernetes build is `make`, so you must
+   have this tool installed on your machine.  We try not to use too many crazy
+   features of `Makefile`s and other tools, so most commonly available versions
+   should work.
+1. `docker` - some parts of the build/test system depend on `docker`.  You
+   need a relatively recent version of Docker installed, and available to you.
```

Comment:
[According to Dan Lorenc](https://groups.google.com/d/msg/kubernetes-sig-cluster-lifecycle/XFODRr_TXgE/hmJ_5v5LCgAJ), minikube is targeting k8s end users. Using it for development of Kubernetes may be [not very convenient](https://github.com/kubernetes/minikube/blob/master/UPDATE_KUBERNETES.md). Or did I miss something here? Also, minikube by itself is probably overkill if the developer just wants to have docker.